src/services/FirebaseAuthService.ts

```diff
@@ -0,0 +1,31 @@
+import firebase from "firebase/app";
+import "firebase/auth";
+import "firebase/firestore";
+
+const firebaseConfig = {
+  apiKey: process.env.FIREBASE_API_KEY,
+  authDomain: process.env.FIREBASE_AUTH_DOMAIN,
+  databaseURL: process.env.FIREBASE_DATABASE_URL,
+  projectId: process.env.FIREBASE_PROJECT_ID,
+  storageBucket: process.env.FIREBASE_STORAGE_BUCKET,
+  messagingSenderId: process.env.FIREBASE_MESSAGING_SENDER_ID
+};
+
+const authenticate = async () => {
+  try {
+    if (!firebase.apps.length) firebase.initializeApp(firebaseConfig);
+
+    const provider = new firebase.auth.GoogleAuthProvider();
+    const result = await firebase.auth().signInWithPopup(provider);
+
+    if (!result.user || !result.credential)
+      throw new Error("Authentication Not Enabled");
```

Comment:
```suggestion
      throw new Error("Authentication Failed");
```

지원하지 않는다기보단, 실패에 가까운 상황 같습니다